### DISCUSSION TITLE
Password entry method, when encrypting/decrypting.

### DISCUSSION THREAD

**Discussion by Articmar** [FiloSottile/age#590] [General]
Please, when encrypting/decrypting with a password, add the ability to pass the password like this `-p <password>`. 
Example:  
`age -e -p <password> -o “path” “path”`, 
`age -d -p <password> -o “path” “path”`.
To make it easier to use the utility in scripts.

**Comment by cipriancraciun** [comment]
Unfortunately, and not because it wasn't asked a million times so far, it appears that `age` would not get soon any sort of password automation feature.  :)

However, for a lengthy discussion about why and possible workarounds, please see discussion #256, and perhaps up-vote it.  :)

**Comment by clach04** [comment]
Cross-reference to https://github.com/FiloSottile/age/discussions/256

@Articmar I recommend closing this as it duplicates existing conversations.

**Comment by mirql** [comment]
using "expect"

https://github.com/FiloSottile/age/pull/520#issuecomment-2768572661

**Comment by FiloSottile** [comment]
The new batchpass plugin handles non-interactive passphrase encryption. See https://github.com/FiloSottile/age/discussions/256#discussioncomment-15334934 for usage and a warning.
